packit_service/service/api/usage.py

```diff
@@ -617,6 +621,92 @@ def get_result_dictionary(
     return {position_name: position, count_name: top_projects.get(project)}
 
 
+@usage_ns.route("/onboarded-projects")
+class Onboarded2024Q1(Resource):
+    @usage_ns.response(
+        HTTPStatus.OK,
+        "Onboarded projects for which exist a Bodhi update or a Koji build or a Packit merged PR.",
+    )
+    @classmethod
+    def calculate(cls):
+        known_onboarded_projects = GitProjectModel.get_known_onboarded_projects()
+
+        bodhi_updates = BodhiUpdateTargetModel.get_all_projects()
+        koji_builds = KojiBuildTargetModel.get_all_projects()
+        onboarded_projects = bodhi_updates.union(koji_builds).union(
+            known_onboarded_projects
+        )
+
+        sync_release_prs_urls = SyncReleaseTargetModel.get_all_downstream_pr_urls()
+        downstream_synced_projects_urls = {
+            pr[0 : pr.rfind("/pull-request")]  # noqa[203] prettier like it this way
+            for pr in sync_release_prs_urls
+        }
+        all_projects = GitProjectModel.get_all()
+        downstream_synced_projects = {
+            project
+            for project in all_projects
+            if project.project_url in downstream_synced_projects_urls
+        }
```

Comment:
I think it may happen that the project from `downstream_synced_projects_urls` is not in our DB, as we store it there probably only if there was some job triggered from dist-git (pull retrigger/koji build/bodhi update) :thinking: (ping @lachmanfrantisek as you were playing with this the last quarter)

so in a situation that dist-git project uses only pull-from-upstream and there was not any retrigger via dist-git comment, I think that project would be not counted with the current implementation

Reply:
Then I am wondering if we have a `GitProjectModel` for a downstream project which has an active and working `pull-from-upstream` job and nothing more; and it has never been retriggered via a comment?

Because I can also think of looping all the downstream `GitProjectModel`s looking for merged PRs (at least once), but now I doubt we have them stored in the db.

Reply:
I was looking at this project: https://src.fedoraproject.org/rpms/speedtest/pull-requests?status=all
I recognize it as onboarded. It has just 3 PR created by Packit and none of them has been retriggered via a comment.
So probably we are safe with this code. What do you think?

Reply:
> Then I am wondering if we have a GitProjectModel for a downstream project which has an active and working pull-from-upstream job and nothing more; and it has never been retriggered via a comment?

I think it is possible that there is no GitProjectModel for such release sync, we could check this in the DB.

Reply:
> I was looking at this project: https://src.fedoraproject.org/rpms/speedtest/pull-requests?status=all The code recognize it as onboarded. It has just 3 PR created by Packit and none of them has been retriggered via a comment. So probably we are safe with this code. What do you think?

Probably in this case it is onboarded because there is a Bodhi update or/and Koji build, since it has all three jobs enabled.

Reply:
I tried using https://sourcegraph.com/search but I got a timeout error.
It can't search through all `src.fedoraproject.org` going back more than 7 months ago, it tells me to refine the search (and in 7 months it finds out 40 projects more or less, against the 222 of the code above). And I can not refine the search if I don't know the downstream git projects names.

So I think we don't have other options,  we need to be sure to have the downstream git project saved and potentially also the `downstream_synced_projects_urls` saved into the db for every sync_release, @lbarcziova @lachmanfrantisek WDYT?

Reply:
discussed with Maja over chat: Maja will check whether all projects obtained via downstream_pr_url are present in GitProjectModel table, I was probably wrong as there are 409 src.fedoraproject.org active projects in prod.packit.dev/usage. 

> Because I can also think of looping all the downstream GitProjectModels looking for merged PRs (at least once), but now I doubt we have them stored in the db.

If that's the case, I agree this might be simpler than getting the projects from downstream_pr_url

Reply:
Sorry for late responses... Yes, it's tricky and exactly the same issues I had.

If we filter out all dist-git URLs, we don't have info for sync retries, Koji and Bodhi. If we don't filter those, we can have duplicates.

Using `downstream_pr_url` is really nice if we don't want to combine this with some upstream info (which we probably don't need).